### ISSUE TITLE
filament::backend::Platform::retrieveBlob on Adreno 506

### ISSUE THREAD

**Issue by victorbrndls** [google/filament#7049]
**Describe the bug**
I thought the `Platform::retrieveBlob` crash was solved in 1.40.4 by `opengl: fix b/290670707 : crash when using the blob cache`, however there are still some users getting this crash.

I noticed that +95% of these crashes happen on Adreno 506 devices.

**To Reproduce**
- Go https://www.lambdatest.com, create a free account
- Select an Adreno 506 device, I used the Moto G6
- Run any app that uses `ModelViewer` to render a model
- Wait a few seconds, it'll crash

**Screenshots**
There should a cube above the arrow but it's rendering nothing
![image_720](https://github.com/google/filament/assets/11446530/c4850f5e-66a0-487b-afbb-a42c831e6d8d)

**Logs**
https://gist.github.com/victorbrndls/d364d1c82e795bed69fcbe59e8028d53
```
Crashed: Thread: SIGSEGV  0x0000000000000038
#00 pc 0x153063 libfilament-jni.so (filament::backend::Platform::retrieveBlob(void const*, unsigned long, void*, unsigned long)) (BuildId: 27cee80154c5ac995e6320e801e5c129bec11fe6)
#01 pc 0x11ba0 libandroid.so (BuildId: 6ca7c5cb01ece365f4aea339728222bc)
#02 pc 0x11bd0 libandroid.so (BuildId: 6ca7c5cb01ece365f4aea339728222bc)
#03 pc 0x11c00 libandroid.so (BuildId: 6ca7c5cb01ece365f4aea339728222bc)
#04 pc 0x11ba0 libandroid.so (BuildId: 6ca7c5cb01ece365f4aea339728222bc)
#05 pc 0x11bd0 libandroid.so (BuildId: 6ca7c5cb01ece365f4aea339728222bc)
#06 pc 0x11c00 libandroid.so (BuildId: 6ca7c5cb01ece365f4aea339728222bc)
#07 pc 0xfc984 libc.so (BuildId: 70c0bef1f8b2d414312145553219d07a)
#08 pc 0x8bdc4 libc.so (BuildId: 70c0bef1f8b2d414312145553219d07a)
#09 pc 0x11ba0 libandroid.so (BuildId: 6ca7c5cb01ece365f4aea339728222bc)
#10 pc 0x11bd0 libandroid.so (BuildId: 6ca7c5cb01ece365f4aea339728222bc)
#11 pc 0x11c00 libandroid.so (BuildId: 6ca7c5cb01ece365f4aea339728222bc)
#12 pc 0x92bac libc.so (BuildId: 70c0bef1f8b2d414312145553219d07a)
#13 pc 0x92b90 libc.so (BuildId: 70c0bef1f8b2d414312145553219d07a)
#14 pc 0x2af2e libc.so (BuildId: 70c0bef1f8b2d414312145553219d07a)
#15 pc 0x92b90 libc.so (BuildId: 70c0bef1f8b2d414312145553219d07a)
#16 pc 0x151548 ld-android.so (BuildId: 26b7e37338cf51d1e52f65c49bd6ad33)
#17 pc 0x269b6 libc.so (BuildId: 70c0bef1f8b2d414312145553219d07a)
```

**Smartphone**
 - Device: Moto G6
 - OS: Android 9

It has also happened on an Oppo A3s.

**Additional context**
Using Filament 1.40.4

**Comment by romainguy** [comment]
@pixelflinger

**Comment by pixelflinger** [comment]
I think `retrieveBlob` is a red herring because `libandroid` is definitely not calling it.

**Comment by sebapian** [comment]
Hi there,

I've also been getting a lot of crashes on phones with GPU Adreno 506. Further details below:

**Smartphone**:

- OPPO CPH1903, OPPO CPH1851, HUAWEI HWDUB-Q, Motorola Moto G (5S) Plus, Phones with GPU Adreno 506
- OS: Android 8.1 (nearly all crashes seem to be from this particular OS so not sure if it plays a part, also looks like highest upgradable Android OS for these devices)

**Filament Version:** 1.34.0

*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
**Log:**
```
backtrace:
  #00  pc 0x000000000099d738  /vendor/lib64/libllvm-glnext.so (llvm::Module::setModuleIdentifier(llvm::StringRef)+156)
  #01  pc 0x000000000099d5f8  /vendor/lib64/libllvm-glnext.so (ESXLinker::bcConstruct()+1172)
  #02  pc 0x000000000099fc18  /vendor/lib64/libllvm-glnext.so (SOLinker::linkShaders(QGLC_LINKPROGRAM_DATA*, QGLC_LINKPROGRAM_RESULT*)+180)
  #03  pc 0x0000000000997298  /vendor/lib64/libllvm-glnext.so (CompilerContext::LinkProgram(unsigned int, QGLC_SRCSHADER_IRSHADER**, QGLC_LINKPROGRAM_DATA*, QGLC_LINKPROGRAM_RESULT*)+492)
  #04  pc 0x0000000000a75370  /vendor/lib64/libllvm-glnext.so (QGLCLinkProgram(void*, unsigned int, QGLC_SRCSHADER_IRSHADER**, QGLC_LINKPROGRAM_DATA*, QGLC_LINKPROGRAM_RESULT*)+104)
  #05  pc 0x00000000001c4ef4  /vendor/lib64/egl/libGLESv2_adreno.so (EsxShaderCompiler::CompileProgram(EsxContext*, EsxProgram const*, EsxLinkedList const*, EsxInfoLog*)+2360)
```

**If this is a seperate issue, I can log another issue. Thank you.**

**Comment by August1996** [comment]
I meet the same problem, and it was solved in 1.42.0

**Comment by August1996** [comment]
> I meet the same problem, and it was solved in 1.41.0我遇到了同样的问题，在1.41.0中解决了

this commit solves the crash.
`workaround a crash with some adreno drivers Mathias Agopian 2023/8/1, 05:41`
